src/rgw/rgw_data_sync.cc

```diff
@@ -1239,17 +1242,16 @@ class RGWDataSyncCR : public RGWCoroutine {
 
       yield {
         if  ((rgw_data_sync_info::SyncState)sync_status.sync_info.state == rgw_data_sync_info::StateSync) {
-          case rgw_data_sync_info::StateSync:
```

Comment:
it looks like this used to be a switch statement that changed to an if on the line above. i assume the compiler didn't complain about the leftover `case` here because the boost `reenter` macro is using a switch under the covers